src/Orckestra.Composer.Cart/Repositories/Order/OrderRepository.cs

```diff
@@ -213,5 +215,43 @@ public virtual Task<HttpWebResponse> DeleteCartOrderDraft(DeleteCartOrderDraftPa
 
             return OvertureClient.SendAsync(request);
         }
+
+        /// <summary>
+        /// Get history items related to specified order
+        /// </summary>
+        /// <param name="param"></param>
+        /// <returns></returns>
+        public virtual Task<OrderFulfillmentState> GetOrderFulfillmentStateAsync(GetOrderFulfillmentStateParam param)
+        {
+            if (param == null) { throw new ArgumentNullException(nameof(param)); }
+            if (string.IsNullOrWhiteSpace(param.ScopeId)) { throw new ArgumentException(GetMessageOfNullWhiteSpace(nameof(param.ScopeId)), nameof(param)); }
+            if (string.IsNullOrWhiteSpace(param.OrderId)) { throw new ArgumentException(GetMessageOfNullWhiteSpace(nameof(param.OrderId)), nameof(param)); }
+
+            var request = new GetOrderFulfillmentStateRequest()
+            {
+                ScopeId = param.ScopeId,
+                OrderId = Guid.Parse(param.OrderId)
+            };
+
+            return OvertureClient.SendAsync(request);
+        }
+
+        public virtual Task<OrderFulfillmentState> ChangeShipmentStatusAsync(ChangeShipmentStatusParam param)
+        {
+            if (param == null) { throw new ArgumentNullException(nameof(param)); }
+            if (string.IsNullOrWhiteSpace(param.ScopeId)) { throw new ArgumentException(GetMessageOfNullWhiteSpace(nameof(param.ScopeId)), nameof(param)); }
+            if (param.OrderId == default) { throw new ArgumentException(GetMessageOfNullWhiteSpace(nameof(param.OrderId)), nameof(param)); }
+
```

Comment:
add validation for RequestedStatus
also check if Reason is required on backend. if yes, also add validation

Reply:
only RequestedStatus is required on backend

Reply:
only RequestedStatus is  required on backend